### PR TITLE
CONCF-743 Use jquery.cropper as a cropping tool in Curated Content Tool

### PR DESCRIPTION
**Pull request by rogatty**
https://wikia-inc.atlassian.net/browse/CONCF-743

### CODE REVIEW

File: front/scripts/main/components/CuratedContentEditorImageSearchComponent.ts [hunk 1]
```diff
@@ -3,20 +3,24 @@
...
+	App.CuratedContentEditorThumbnailMixin,
 	App.LoadingSpinnerMixin,
 	App.TrackClickMixin,
 	{
 		classNames: ['curated-content-editor-image-search'],
 		debounceDuration: 300,
+		imageWidth: 200,
+		imageHeight: 200,
+		searchLimit: 24,
+		nextBatch: 1,
```
Comment (rogatty) [inline front/scripts/main/components/CuratedContentEditorImageSearchComponent.ts]: Those 4 lines shouldn't be here.

File: front/scripts/main/components/CuratedContentEditorImageCropComponent.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+		initCropper(): void {
+			var $imgElement = this.$(this.get('imgSelector')),
+				settings: any = $.extend(this.get('cropperSettings'), {
+					aspectRatio: this.get('aspectRatio')
+				});
+
+			$imgElement.cropper(settings);
+
+			this.setProperties({
```
Comment (hakubo) [inline front/scripts/main/components/CuratedContentEditorImageCropComponent.ts]:
```
this.setProperties({
  isLoading: false,
  $imgElement
});
```

File: front/scripts/main/components/CuratedContentEditorItemFormComponent.ts [hunk 3]
```diff
@@ -14,12 +14,15 @@ App.CuratedContentEditorItemFormComponent = Em.Component.extend(
...
-		imageSize: 300,
+		imageWidth: 300,
 		maxLabelLength: 48,
 		debounceDuration: 250,
 
-		imageUrl: Em.computed('model.image_url', function (): string {
-			return this.generateThumbUrl(this.get('model.image_url'));
+		imageUrl: Em.computed('model.image_url', 'model.image_crop', function (): string {
+			var aspectRatioName = this.get('aspectRatioName'),
+				imageCrop = this.get('model.image_crop.' + aspectRatioName) || null;
```
Comment (hakubo) [inline front/scripts/main/components/CuratedContentEditorItemFormComponent.ts]:
```
`model.image_crop.${aspectRatioName}`
```

Comment (rogatty) [inline front/scripts/main/components/CuratedContentEditorItemFormComponent.ts]: This is modified in CONCF-977 anyway.

File: front/scripts/main/components/CuratedContentEditorImageCropComponent.ts (hunk 2 above)
Comment (tomasznapieralski) [inline front/scripts/main/components/CuratedContentEditorImageCropComponent.ts]: fixed

**Comment by hakubo** [comment]
LGTM

File: front/scripts/main/mixins/CuratedContentEditorLayoutMixin.ts [hunk 4]
```diff
@@ -2,12 +2,24 @@
...
+		id: ''
+	},
+
 	itemFormLayout: {
 		name: 'curated-content-editor-item-form'
 	},
+
 	imageSearchLayout: {
 		name: 'curated-content-editor-image-search',
 		next: 'curated-content-editor-item-form',
```
Comment (rogatty) [inline front/scripts/main/mixins/CuratedContentEditorLayoutMixin.ts]: This should be `curated-content-editor-image-crop`

**Pull request merged**